Problem3/Analysis.java

```diff
@@ -40,6 +40,22 @@ public void log(String msg)
 	}
 }
 
+class LoggingFactory
+{
+
+	public static Logging createLogging(String type){
+		if (type.equalsIgnoreCase("text"))
```

Comment:
What are your thoughts on the unused enum LOG_TYPE defined in the interface?  I think we might be able to remove it.